AppCenterDistribute/AppCenterDistribute/MSACDistribute.m

```diff
@@ -144,6 +144,12 @@ - (instancetype)init {
 
     // Init the distribute info tracker.
     _distributeInfoTracker = [[MSACDistributeInfoTracker alloc] init];
+
+    // In case we never finished removing this key. Which will result in
+    // update flow never gets processed again
+    if ([MSAC_APP_CENTER_USER_DEFAULTS objectForKey:kMSACUpdateTokenRequestIdKey]) {
+      [MSAC_APP_CENTER_USER_DEFAULTS removeObjectForKey:kMSACUpdateTokenRequestIdKey];
```

Comment:
Need to be covered by some unit test. @Petr-2019 I'm just pointing this out, the team can take care of finishing the change.

Reply:
100% agree, thanks for pointing it out.